x/incentives/keeper/gauge_test.go

```diff
@@ -1114,6 +1114,101 @@ func (s *KeeperTestSuite) TestChargeGroupCreationFeeIfNotWhitelisted() {
 	}
 }
 
+func (s *KeeperTestSuite) TestGetPoolIdsAndDurationsFromGroup() {
+	s.SetupTest()
+
+	pools := s.PrepareAllSupportedPools()
+
+	testCases := []struct {
+		name              string
+		expectedPoolIds   []uint64
+		expectedDurations []time.Duration
+	}{
+		{
+			name:              "Two pool IDs of various types",
+			expectedPoolIds:   []uint64{pools.BalancerPoolID, pools.StableSwapPoolID},
+			expectedDurations: []time.Duration{time.Hour * 7, time.Hour * 7},
+		},
+		{
+			name:              "Three pool IDs of various types",
+			expectedPoolIds:   []uint64{pools.ConcentratedPoolID, pools.BalancerPoolID, pools.StableSwapPoolID},
+			expectedDurations: []time.Duration{time.Hour * 24 * 7, time.Hour * 7, time.Hour * 7},
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(tc.name, func() {
+			equalPoolVolumes, _ := setupEqualVolumeWeights(len(tc.expectedPoolIds), oneMillionVolumeAmt)
+			s.SetupVolumeForPools(tc.expectedPoolIds, equalPoolVolumes, map[uint64]osmomath.Int{})
+
+			gaugeId, err := s.App.IncentivesKeeper.CreateGroup(s.Ctx, defaultCoins, types.PerpetualNumEpochsPaidOver, s.TestAccs[0], tc.expectedPoolIds)
+			s.Require().NoError(err)
+
+			group, err := s.App.IncentivesKeeper.GetGroupByGaugeID(s.Ctx, gaugeId)
+			s.Require().NoError(err)
+
+			// system under test
+			poolIds, durations, err := s.App.IncentivesKeeper.GetPoolIdsAndDurationsFromGroup(s.Ctx, group)
+			s.Require().NoError(err)
+
+			s.Require().Equal(tc.expectedPoolIds, poolIds)
+			s.Require().Equal(tc.expectedDurations, durations)
+
+			// checks the underlying helper function as well
+			for i, gaugeRecord := range group.InternalGaugeInfo.GaugeRecords {
+				poolId, duration, err := s.App.IncentivesKeeper.GetPoolIdAndDurationFromGaugeRecord(s.Ctx, gaugeRecord)
+				s.Require().NoError(err)
+				s.Require().Equal(tc.expectedPoolIds[i], poolId)
+				s.Require().Equal(tc.expectedDurations[i], duration)
+			}
```

Comment:
Given that we test `GetPoolIdAndDurationFromGaugeRecord` via its own fixture, we can remove this to keep this test focused on the actual system under test - `GetPoolIdsAndDurationsFromGroup`

Reply:
https://github.com/osmosis-labs/osmosis/pull/6601/commits/9bc2b71b228ef699d39ac843afaad5efbf3342ec